tests/python/pants_test/BUILD

```diff
@@ -58,6 +58,15 @@ python_library(
     'src/python/pants/util:process_handler',
     'tests/python/pants_test/testutils:file_test_util',
     'tests/python/pants_test/testutils:pexrc_util',
+    # NB: Integration tests in the pants codebase find the pants script by absolute path, which
+    # escapes the chroot which tests are executed in, allowing them to use files they don't
+    # explicitly depend on. This dependency forces every test depending on this target to depend on
+    # ~all the pants source, so the result is invalidated when changing any relevant source files. If
+    # this is removed, the result of integration tests will be cached regardless of changes to
+    # source files that affect their behavior. This does not occur in CI due to the use of
+    # clean-all.
+    # There may be a better way to do this.
+    'src/python/pants/bin:pants_local_binary',
```

Comment:
Hrm. Forgot about `:test_infra` above. It should probably depend on the original version of this target and then all other `python_tests` targets in the repo depending on `:int-test` should depend on the enhanced version that adds in the new dep. For the smallest ripple, I'd suggest re-naming `:int-test` to, say `:int-test-for-export`, then adding a:
```
target(
  name = 'int-test',
  dependencies=[
    ':int-test-for-export',
    # All the comment ...
    'src/python/pants/bin:pants_local_binary',
  ]
)
```

Reply:
That makes sense -- I hadn't seen your comment until after I just pushed. I'll make the change.